src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImpl.java

```diff
@@ -0,0 +1,190 @@
+import org.kuali.kfs.sys.document.AccountingDocument;
+import org.kuali.kfs.sys.service.FileStorageService;
+import org.kuali.rice.core.api.config.property.ConfigurationService;
+import org.kuali.rice.kew.api.exception.WorkflowException;
+import org.springframework.util.AutoPopulatingList;
+
+import edu.cornell.kfs.fp.CuFPConstants;
+import edu.cornell.kfs.fp.batch.service.AccountingDocumentGenerator;
+import edu.cornell.kfs.fp.batch.service.CreateAccountingDocumentService;
+import edu.cornell.kfs.fp.batch.xml.AccountingXmlDocumentEntry;
+import edu.cornell.kfs.fp.batch.xml.AccountingXmlDocumentListWrapper;
+
+public class CreateAccountingDocumentServiceImpl implements CreateAccountingDocumentService {
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(CreateAccountingDocumentServiceImpl.class);
+
+    private BatchInputFileService batchInputFileService;
+    private BatchInputFileType accountingDocumentBatchInputFileType;
+    private DocumentService documentService;
+    private FileStorageService fileStorageService;
+    private ConfigurationService configurationService;
+
+    @Override
+    public void createAccountingDocumentsFromXml() {
+        List<String> inputFileNames = batchInputFileService.listInputFileNamesWithDoneFile(accountingDocumentBatchInputFileType);
+        LOG.info("createAccountingDocumentsFromXml: Found " + inputFileNames.size() + " files to process");
+        
+        inputFileNames.stream()
+                .forEach(this::processAccountingDocumentFromXml);
+        
+        LOG.info("createAccountingDocumentsFromXml: Finished processing all pending accounting document XML files");
+    }
+
+    protected void processAccountingDocumentFromXml(String fileName) {
+        try {
+            LOG.info("processAccountingDocumentFromXml: Started processing accounting document XML file: " + fileName);
+            
+            byte[] fileData = safelyLoadFileBytes(fileName);
+            AccountingXmlDocumentListWrapper accountingXmlDocuments = (AccountingXmlDocumentListWrapper) batchInputFileService.parse(
+                    accountingDocumentBatchInputFileType, fileData);
+            int documentCount = accountingXmlDocuments.getDocuments().size();
+            LOG.info("processAccountingDocumentFromXml: Found " + documentCount + " documents to process from file: " + fileName);
+            accountingXmlDocuments.getDocuments().stream()
+                    .forEach(this::processAccountingDocumentEntryFromXml);
+            
+            LOG.info("processAccountingDocumentFromXml: Finished processing accounting document XML file: " + fileName);
+        } catch (Exception e) {
+            LOG.error("processAccountingDocumentFromXml: Error processing accounting document XML file", e);
+        } finally {
+            removeDoneFileQuietly(fileName);
+        }
+    }
+
+    protected byte[] safelyLoadFileBytes(String fileName) {

```

Comment:
I am not sure if this is the right place to do this, but the function safelyLoadFileBytes exists in multiple locations in our code base.  It would be nice to move that to some utility class.

Reply:
I agree that we need a better common spot for methods like this, but I think that such a change is beyond the scope of this story. I'll create a new user story for doing so.

Reply:
sounds good, thanks

Reply:
I created KFSPTS-10437 for cleanin up safelyLoadFileBytes